walletdb/bdb/db.go

```diff
@@ -48,34 +48,88 @@ func convertErr(err error) error {
 	return err
 }
 
+// transaction represents a database transaction.  It can either by read-only or
+// read-write and implements the walletdb Tx interfaces.  The transaction
+// provides a root bucket against which all read and writes occur.
+type transaction struct {
+	boltTx *bolt.Tx
+}
+
+func (tx *transaction) ReadBucket(key []byte) walletdb.ReadBucket {
+	return tx.ReadWriteBucket(key)
+}
+
+func (tx *transaction) ReadWriteBucket(key []byte) walletdb.ReadWriteBucket {
+	boltBucket := tx.boltTx.Bucket(key)
+	if boltBucket == nil {
+		return nil
+	}
+	return (*bucket)(boltBucket)
+}
+
+func (tx *transaction) CreateTopLevelBucket(key []byte) (walletdb.ReadWriteBucket, error) {
```

Comment:
Going off of the `godoc` comment in `interface.go`, the line below should actually be using `CreateBucketIfNotExists`.